xdsl/dialects/pdl.py

```diff
@@ -0,0 +1,322 @@
+                  covariant=True)
+
+
+@irdl_attr_definition
+class RangeType(Generic[_RangeT], ParametrizedAttribute, MLIRType):
+    name = "pdl.range"
+    element_type: ParameterDef[_RangeT]
+
+
+@irdl_op_definition
+class ApplyNativeConstraintOp(Operation):
+    """
+    https://mlir.llvm.org/docs/Dialects/PDLOps/#pdlapply_native_constraint-mlirpdlapplynativeconstraintop
+    """
+    name: str = "pdl.apply_native_constraint"
+    # https://github.com/xdslproject/xdsl/issues/98
+    # name: OpAttr[StringAttr]
+    args: Annotated[VarOperand, AnyPDLType]
+
+    def verify_(self) -> None:
+        if 'name' not in self.attributes:
+            raise Exception(
+                "ApplyNativeConstraintOp requires a 'name' attribute")
+
+        if not isinstance(self.attributes['name'], StringAttr):
+            raise Exception("expected 'name' attribute to be a StringAttr")
+
+    @staticmethod
+    def get(name: str, args: list[SSAValue]) -> ApplyNativeConstraintOp:
+        return ApplyNativeConstraintOp.build(
+            result_types=[],
+            operands=[args],
+            attributes={"name": StringAttr(name)})
+
+
+@irdl_op_definition
+class ApplyNativeRewriteOp(Operation):
+    """
+    https://mlir.llvm.org/docs/Dialects/PDLOps/#pdlapply_native_rewrite-mlirpdlapplynativerewriteop
+    """
+    name: str = "pdl.apply_native_rewrite"
+    # https://github.com/xdslproject/xdsl/issues/98
+    # name: OpAttr[StringAttr]
+    args: Annotated[VarOperand, AnyPDLType]
+    res: Annotated[VarOpResult, AnyPDLType]
+
+    def verify_(self) -> None:
+        if 'name' not in self.attributes:
+            raise Exception("ApplyNativeRewriteOp requires a 'name' attribute")
+
+        if not isinstance(self.attributes['name'], StringAttr):
+            raise Exception("expected 'name' attribute to be a StringAttr")
+
+    @staticmethod
+    def get(name: str, args: list[SSAValue],
+            result_types: list[Attribute]) -> ApplyNativeRewriteOp:
+
+        return ApplyNativeRewriteOp.build(
+            result_types=[result_types],
+            operands=[args],
+            attributes={"name": StringAttr(name)})
+
+
+@irdl_op_definition
+class AttributeOp(Operation):
+    """
+    https://mlir.llvm.org/docs/Dialects/PDLOps/#pdlattribute-mlirpdlattributeop
+    """
+    name: str = "pdl.attribute"
+    value: OptOpAttr[Attribute]
+    value_type: Annotated[OptOperand, TypeType]
+    output: Annotated[OpResult, AttributeType]
+
+
+@irdl_op_definition
+class EraseOp(Operation):
+    """
+    https://mlir.llvm.org/docs/Dialects/PDLOps/#pdlerase-mlirpdleraseop
+    """
+    name: str = "pdl.erase"
+    op_value: Annotated[Operand, OperationType]
+
+
+@irdl_op_definition
+class OperandOp(Operation):
+    """
+    https://mlir.llvm.org/docs/Dialects/PDLOps/#pdloperand-mlirpdloperandop
+    """
+    name: str = "pdl.operand"
+    value_type: Annotated[OptOperand, TypeType]
+    output: Annotated[OpResult, ValueType]
```

Comment:
Is there a specific reason why you renamed this?

Reply:
Ah this I inherited, I'll try renaming
- [ ] rename output to value in `pdl.operand` and `pdl.operands`